gio/src/settings.rs

```diff
@@ -23,6 +35,74 @@ impl<O: IsA<Settings>> SettingsExtManual for O {
     fn set<U: ToVariant>(&self, key: &str, value: &U) -> Result<(), BoolError> {
         self.set_value(key, &value.to_variant())
     }
+
+    fn bind_with_mapping<P: IsA<glib::Object>>(
+        &self,
+        key: &str,
+        object: &P,
+        property: &str,
+        flags: SettingsBindFlags,
+        get_mapping: Option<Box<dyn Fn(&glib::Variant, glib::Type) -> Option<glib::Value>>>,
+        set_mapping: Option<Box<dyn Fn(&glib::Value, glib::VariantType) -> Option<glib::Variant>>>,
+    ) {
+        type Mappings = (
+            Option<Box<dyn Fn(&glib::Variant, glib::Type) -> Option<glib::Value>>>,
+            Option<Box<dyn Fn(&glib::Value, glib::VariantType) -> Option<glib::Variant>>>,
+        );
+        unsafe extern "C" fn bind_with_mapping_get_trampoline(
+            value: *mut glib::gobject_ffi::GValue,
+            variant: *mut glib::ffi::GVariant,
+            user_data: glib::ffi::gpointer,
+        ) -> glib::ffi::gboolean {
+            let user_data = &*(user_data as *const Mappings);
+            let f = user_data.0.as_ref().unwrap();
+            let value = &mut *(value as *mut glib::Value);
+            if let Some(v) = f(&*from_glib_borrow(variant), value.type_()) {
```

Comment:
So the value is initialized with the correct target type?

Reply:
Yep.

> GSettingsBindGetMapping  
> The type for the function that is used to convert from GSettings to an object property. The value is already initialized to hold values of the appropriate type.

https://developer.gnome.org/gio/stable/GSettings.html

Reply:
You could then even put a Rust panic at the end of the trampoline if you want

Reply:
I.e. `assert!(v.type_().is_a(value.type_()))`

Reply:
Let me know if you want to do something like that or just keep it as is. If you do, should maybe do the same for the other callback too

Reply:
Hm. I guess there's a somewhat broader question of whether gtk-rs should panic in cases like this, where gtk prints a warning but keeps working.

The helpful warning glib already print seems good enough, given it can't just be enforced by the type system (now if only Rust had full dependent typing...).

Reply:
I guess it's fine if it doesn't cause things to misbehave later. Let's keep it like this for now then.